<a href="https://colab.research.google.com/github/HibaANBusttami-April/HibaANBusttami-April/blob/main/Copy_of_Prediction_of_Product_Sales_HIBA_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Prediction of Product Sales
- Hiba A. N. Busttami (April)


## Project Overview


## Load and Inspect Data

In [ ]:
from google.colab import drive
drive.mount('/content/drive')
# Import required packages
import pandas as pd

Mounted at /content/drive


In [ ]:
# Save the filepath and load the data as "df"
data="/content/drive/MyDrive/CodingDojo/01-Fundamentals/Week02/Data/sales_predictions_2023.csv"
# read the data
df= pd.read_csv(data)
# Display the first few rows of the dataframe to ensure it's loaded correctly
df.head()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


## Clean Data

-  **How many rows and columns?**
 - We have 8523 rows and 12 columns

In [ ]:
# Get the shape of the DataFrame
df_shape = df.shape
print("Shape of the DataFrame:", df_shape)


Shape of the DataFrame: (8523, 12)


In [ ]:
# Get the datatypes of each column
column_dtypes = df.dtypes
print("Datatypes of each column:")
print(column_dtypes)


Datatypes of each column:
Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object


- **What are the datatypes of each variable?**

 - **Item_Identifier**  :               *object*
 - **Item_Weight**  :                  *float64*
 - **Item_Fat_Content**  :              *object*
 - **Item_Visibility**  :              *float64*
 - **Item_Type**  :                     *object*
 - **Item_MRP**  :                     *float64*
 - **Outlet_Identifier**  :             *object*
 - **Outlet_Establishment_Year**  :      *int64*
 - **Outlet_Size**  :                   *object*
 - **Outlet_Location_Type**  :          *object*
 - **Outlet_Type**  :                   *object*
 - **Item_Outlet_Sales**  :            *float64*

- Are there duplicates? If so, drop any duplicates ?

In [ ]:
# Check for duplicate rows
duplicates = df.duplicated().sum()
print("Number of duplicate rows:", duplicates)
# Drop duplicate rows
df_cleaned = df.drop_duplicates()

# Verify that duplicates have been removed
new_duplicates = df_cleaned.duplicated().sum()
print("Number of duplicate rows after dropping:", new_duplicates)



Number of duplicate rows: 0
Number of duplicate rows after dropping: 0


- Identify missing values.

In [ ]:
# Check for missing values in each column
missing_values = df.isnull().sum()
print("Missing values in each column:")
print(missing_values)


Missing values in each column:
Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64


In [ ]:
# List of numeric column names
numeric_columns = df.select_dtypes(include=['number']).columns.tolist()
print(numeric_columns)

['Item_Weight', 'Item_Visibility', 'Item_MRP', 'Outlet_Establishment_Year', 'Item_Outlet_Sales']


In [ ]:
#calculate the mean to use it as a place holder in Item weight column
mean=df["Item_Weight"].mean()
df["Item_Weight"] = df["Item_Weight"].fillna(mean)
# Verify that null values have been filled
print(df[numeric_columns].isnull().sum())

Item_Weight                  0
Item_Visibility              0
Item_MRP                     0
Outlet_Establishment_Year    0
Item_Outlet_Sales            0
dtype: int64


In [ ]:
# Get a list of categorical column names
categorical_columns = df.select_dtypes(include=['object']).columns.tolist()

# Print the list of categorical columns
print(categorical_columns)

['Item_Identifier', 'Item_Fat_Content', 'Item_Type', 'Outlet_Identifier', 'Outlet_Size', 'Outlet_Location_Type', 'Outlet_Type']


In [ ]:
# Fill null values in categorical columns with 'MISSING'
df[categorical_columns] = df[categorical_columns].fillna('MISSING')

# Verify that null values have been filled
print(df[categorical_columns].isnull().sum())

Item_Identifier         0
Item_Fat_Content        0
Item_Type               0
Outlet_Identifier       0
Outlet_Size             0
Outlet_Location_Type    0
Outlet_Type             0
dtype: int64


 - Find and fix any inconsistent categories of data (for example: fix cat, Cat, and cats so that they are consistent).

In [ ]:
# Iterate through each column and print unique values
for column in df.columns:
    value_counts = df[column].value_counts()
    print(f"Unique values in column '{column}':")
    print(value_counts)


Unique values in column 'Item_Identifier':
Item_Identifier
FDW13    10
FDG33    10
NCY18     9
FDD38     9
DRE49     9
         ..
FDY43     1
FDQ60     1
FDO33     1
DRF48     1
FDC23     1
Name: count, Length: 1559, dtype: int64
Unique values in column 'Item_Weight':
Item_Weight
-1.000     1463
 12.150      86
 17.600      82
 13.650      77
 11.800      76
           ... 
 7.275        2
 7.685        1
 9.420        1
 6.520        1
 5.400        1
Name: count, Length: 416, dtype: int64
Unique values in column 'Item_Fat_Content':
Item_Fat_Content
Low Fat    5089
Regular    2889
LF          316
reg         117
low fat     112
Name: count, dtype: int64
Unique values in column 'Item_Visibility':
Item_Visibility
0.000000    526
0.076975      3
0.162462      2
0.076841      2
0.073562      2
           ... 
0.013957      1
0.110460      1
0.124646      1
0.054142      1
0.044878      1
Name: count, Length: 7880, dtype: int64
Unique values in column 'Item_Type':
Item_Type
Fruits and Veg

In [ ]:
# Define the data as a dictionary
newfat_content_dict = {
    'LF': 'Low Fat',
    'reg': 'Regular',
    'low fat':'Low Fat'
}
# Replace inconsistent values using the dictionary
df['Item_Fat_Content'] = df['Item_Fat_Content'].replace(newfat_content_dict)
# Verify the changes
print("DataFrame after standardizing 'Fat_Content':")
value_countsfat = df["Item_Fat_Content"].value_counts()
print(value_countsfat)

DataFrame after standardizing 'Fat_Content':
Item_Fat_Content
Low Fat    5517
Regular    3006
Name: count, dtype: int64


In [ ]:
# Get summary statistics for numerical columns
summary_stats = df.describe(include=[float, int])
print("Summary statistics for numerical columns:")
print(summary_stats)


Summary statistics for numerical columns:
       Item_Weight  Item_Visibility     Item_MRP  Outlet_Establishment_Year  \
count  8523.000000      8523.000000  8523.000000                8523.000000   
mean     10.478936         0.066132   140.992782                1997.831867   
std       6.720742         0.051598    62.275067                   8.371760   
min      -1.000000         0.000000    31.290000                1985.000000   
25%       6.650000         0.026989    93.826500                1987.000000   
50%      11.000000         0.053931   143.012800                1999.000000   
75%      16.000000         0.094585   185.643700                2004.000000   
max      21.350000         0.328391   266.888400                2009.000000   

       Item_Outlet_Sales  
count        8523.000000  
mean         2181.288914  
std          1706.499616  
min            33.290000  
25%           834.247400  
50%          1794.331000  
75%          3101.296400  
max         13086.964800  


## Exploratory Data Analysis

## Feature Inspection